### PR TITLE
Package ocaml-protoc.2.0.1

### PR DESCRIPTION
**Pull request by mransan**
### `ocaml-protoc.2.0.1`
Protobuf compiler for OCaml
Protobuf compiler for OCaml

---
* Homepage: https://github.com/mransan/ocaml-protoc
* Source repo: git+https://github.com/mransan/ocaml-protoc.git
* Bug tracker: https://github.com/mransan/ocaml-protoc/issues

---
:camel: Pull-request generated by opam-publish v2.0.0

### CODE REVIEW

File: packages/ocaml-protoc/ocaml-protoc.2.0.1/opam [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+bug-reports:"https://github.com/mransan/ocaml-protoc/issues"
+dev-repo:"git+https://github.com/mransan/ocaml-protoc.git"
+license: "MIT"
+build: [
+  ["dune" "subst"] {pinned}
+  ["dune" "build" "-p" name "-j" jobs]
+]
+depends: [
+  "ocaml" {>="4.02.1"}
+  "dune"  {build}
```
Comment (kit-ty-kate) [inline packages/ocaml-protoc/ocaml-protoc.2.0.1/opam]:
```suggestion
  "dune"  {>= "1.11"}
```
Dune should not be tagged as a build dependency (see #14266). Moreover your `dune-project` file says this package requires at least dune 1.11

File: packages/ocaml-protoc/ocaml-protoc.2.0.1/opam [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  "stdlib-shims"
+  "ocamlbuild" {build}
```
Comment (kit-ty-kate) [inline packages/ocaml-protoc/ocaml-protoc.2.0.1/opam]:
```suggestion
```
This packages doesn't seem to depend on ocamlbuild anymore

Comment (mransan) [reply packages/ocaml-protoc/ocaml-protoc.2.0.1/opam]: unfortunately, the following [package](https://opam.ocaml.org/packages/ocamlbuild-protoc/) has a transitive dependency on ocamlbuild through ocaml-protoc. Here is the  [issue](https://github.com/Khady/ocamlbuild-protoc/issues/1). The short term fix is to leave the ocamlbuild dependency in this package pending the resolution. 

(see my previous publishing attempt [here](https://github.com/ocaml/opam-repository/pull/15204)

File: packages/ocaml-protoc/ocaml-protoc.2.0.1/opam (hunk 1 above)
Comment (mransan) [reply packages/ocaml-protoc/ocaml-protoc.2.0.1/opam]: Thanks for the suggestion and info. I'll accept it.

**Comment by mransan** [comment]
@kit-ty-kate thanks for looking at this. I accept your suggestion for the dune specification but due to an unfortunate issue with a reverse dependency I cannot get rid of the ocamlbuild build dependency like we both would like to :(

**Comment by kit-ty-kate** [comment]
cc @Khady for the change in `ocamlbuild-protoc`

**Comment by kit-ty-kate** [comment]
> @kit-ty-kate thanks for looking at this. I accept your suggestion for the dune specification but due to an unfortunate issue with a reverse dependency I cannot get rid of the ocamlbuild build dependency like we both would like to :(

yes it can be done. Other packages can be modified/fixed if needed. Anyone can request a change

**Comment by mransan** [comment]
@kit-ty-kate  this is awesome. thanks for making the fix.

**Comment by kit-ty-kate** [comment]
Thanks

**Pull request merged**